### PR TITLE
fix `istioctl x wait` command not works with resourceversion's length longer than 8

### PR DESCRIPTION
**Pull request by cocotyty**
Please provide a description for what this PR is for.

Resourceversion in my k8s cluster is longer than 8  , like `549779490`.

When put resourceversion into ledger ( [code](https://github.com/istio/pkg/blob/6775f12cf1002997b00eea359016ae533587d5c6/ledger/ledger.go#L57) ), it corce to hash length (length is 8).
So wait command not works in my cluster, it always print 
```
Error: timeout expired before resource VirtualService/**/**** became effective on all sidecars
```
I add a method called containsVersion , it fix this problem.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[X] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by cocotyty** [comment]
@howardjohn  @therealmitchconnors @zhaohuabing

**Comment by esnible** [comment]
/ok-to-test

@therealmitchconnors Can you review this?

**Comment by cocotyty** [comment]
```log
2020-12-21T03:49:49.809043Z	error	tf	Unable to get istio-proxy config dump for pod: echo-1-82806/tproxy-v1-78fd647545-l4bxz
2020-12-21T03:49:49.876924Z	error	tf	Unable to get istio-proxy clusters for pod: echo-1-82806/tproxy-v1-78fd647545-l4bxz
2020-12-21T03:49:53.837539Z	error	tf	=== Dumping Namespace external-2-68919 State...
2020-12-21T03:49:54.284069Z	info	tf	=== BEGIN: Cleanup Istio [Suite=pilot_endpointslice] ===
2020-12-21T03:49:55.434147Z	info	tf	=== SUCCEEDED: Cleanup Istio in 1.149991433s [Suite=pilot_endpointslice] ===
FAIL	istio.io/istio/tests/integration/pilot/endpointslice	69.590s
```
@howardjohn @esnible  I don't know why.

**Comment by cocotyty** [comment]
@therealmitchconnors please review this pr, thanks . 😊

File: releasenotes/notes/29678.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: istioctl
+issue:
+
+releaseNotes:
+  - |
+    **Fixed** a bug that `istioctl x wait` can't works with resourceVersion length longer than 8.
```
Comment (ericvn) [inline releasenotes/notes/29678.yaml:8]:
```suggestion
    **Fixed** an issue where `istioctl x wait` times out when the resourceVersion length is longer than 8.
```

Comment (cocotyty) [reply releasenotes/notes/29678.yaml:8]: Thanks!

**Comment by therealmitchconnors** [comment]
Hi @cocotyty thanks for the PR!

I think this issue is resolved in https://github.com/istio/pkg/pull/278, which I am trying to get merged before the 1.9 code complete date.  This will allow istio to track full CRD generations, regardless of length.  If there's something I've missed, please let me know.

**Pull request closed without merging**